lib/motion-spec/core.rb

```diff
@@ -26,67 +26,75 @@ module MotionSpec
     'colorized' => ColorizedOutput
   }
 
-  def self.add_context(context)
-    (@contexts ||= []) << context
-  end
-
-  def self.current_context_index
-    @current_context_index ||= 0
-  end
+  class << self
+    def add_context(context)
+      (@contexts ||= []) << context
+    end
 
-  def self.current_context
-    @contexts[current_context_index]
-  end
+    def current_context_index
+      @current_context_index ||= 0
+    end
 
-  def self.run(arg = nil)
-    unless respond_to?(:handle_specification_begin)
-      extend(Outputs[ENV['output']] || DEFAULT_OUTPUT_MODULE)
+    def current_context
+      @contexts[current_context_index]
     end
 
-    @timer ||= Time.now
+    def run(arg = nil)
+      set_default_output
 
-    if Platform.android?
-      @main_activity ||= arg
+      @timer ||= Time.now
 
-      @contexts.each { |context| execute_context(context) }
-      return handle_summary
+      Counter[:context_depth] += 1
+      Platform.android? ? run_android_specs : run_cocoa_specs
     end
 
-    Counter[:context_depth] += 1
-    handle_specification_begin(current_context.name)
-    current_context.performSelector('run', withObject: nil, afterDelay: 0)
-  end
-
-  def self.execute_context(context)
-    unless respond_to?(:handle_specification_begin)
-      extend(Outputs[ENV['output']] || DEFAULT_OUTPUT_MODULE)
+    # Android-only.
+    def main_activity
+      @main_activity
     end
 
-    Counter[:context_depth] += 1
-    handle_specification_begin(context.name)
-    context.run
-    handle_specification_end
-    Counter[:context_depth] -= 1
-  end
+    def context_did_finish(context)
```

Comment:
Unused method argument - `context`. If it's necessary, use `_` or `_context` as an argument name to indicate that it won't be used. You can also write as `context_did_finish(*)` if you want the method to accept any arguments but don't care about them.